### PR TITLE
Add: New disaster mode that enables all disaster types regardless of era

### PR DESCRIPTION
**Pull request by SamuXarick**
This mode allows all disaster types to happen regardless of game year.

There is a small problem when loading the old openttd.cfg file. Not sure how to go about it.
screenshot

The workaround is to manually re-set the disaster setting and exit openttd to update the .cfg file.

### CODE REVIEW

**Comment by TrueBrain** [comment]
On the balance of things, I do not think this fits in vanilla OpenTTD. A few reasons:

1) in my observation, only very few people play with disasters, and for good reason. It is a "fun" gimmick, but only for a single playthrough. These days games add "disasters" differently, by being non-destructive but still annoying. Take for example the recently released Creeper World 4 (excellent game, and he is not even paying me to say that): in that game one of the "disasters" temporary disables your buildings. That is a good example of a non-destructive disaster: still annoying, but you will be fine. In a game like OpenTTD is having your rails destroyed on the other end of the map just .. not fun. And although this is a personal opinion of mine, it weights in on the balance if we should even have a settings: "do disasters all the time".
2) disasters in eras are tuned to be ... not there all the time. I cannot see how the gameplay improves by having it happening all the time. You also supply no reasoning or intention for this change, which doesn't convince me otherwise.
3) over the last decade we have slowed down with the amount of settings, nobs, and switches people can pull. Changing a simple: disasters yes/no to a more complex: yes, sometimes, always, is adding complexity I do not see a reason for, even not for a very few that might be using it.
4) given there is no evidence to the contrary, and you are literally the first person I see doing anything with disasters in 10 years, I think you and you alone have this itch. I tend to be hesitant towards features that only scratch the itch of 1 person.

All in all, enough reason for me to close this. Of course all these points can be debated, as it is an opinion. But debate happens with reasoning, so the only way I am willing to reconsider is when there is some carefully thought-out reasoning this is a good idea after all, and show that at least more than 2 people support the idea :)

tldr; this Pull Request is missing any motivation why this should be done; after some careful thinking, I cannot come to a valid reasoning to merge this Pull Request.
